### ISSUE TITLE
Allow overriding the severity of all analyzer rules in Ruleset files

### ISSUE THREAD

**Issue by ohadschn** [dotnet/roslyn#20310]
Suppose I want to treat all issues in the [SonarAnalyzer.CSharp](https://www.nuget.org/packages/SonarAnalyzer.CSharp/) Roslyn analyzer as errors. Visual Studio provides a convenient interface for that (allowing to set the severity of an entire analyzer), but the resulting ruleset file simply has all the analyzer's rules specifically configured as errors one by one (`<Rule Id="..." Action="Error" />`).

The problem with that is that the next time I update the SonarAnalyzer.Csharp NuGet, some new rules will get added and I won't have them show up as errors as well without updating the ruleset file. It would be much more convenient (and easier to maintain) if we had some sort of `OverrideSeverity` option, say as an attribute on the `Rules` element. Specific `Rule` elements will still have the final word of course.

Then I could write something like this:

``` xml
<RuleSet Name="Foo" Description="Bar" ToolsVersion="15.0">
  <IncludeAll Action="Error" />
  <Rules AnalyzerId="SonarAnalyzer" RuleNamespace="SonarAnalyzer" OverrideSeverity="Error">
    <Rule Id="S2330" Action="Warning" />
    <Rule Id="S1121" Action="None" />
  </Rules>
</RuleSet>
```

And now all `SonarAnalyzer` rules (at least those enabled by default) will be treated as errors, except `S2330` which will be a warning, and `S1121` which will be suppressed.

**Comment by sharwell** [comment]
The rule set files have proven limiting in a number of ways. What we'd like to do is design an expansion of the .editorconfig support to cover additional use cases like this, but it's on the back burner while we get basic analyzer configuration features working with it. The one issue we've filed so far related to this is #19026.

**Comment by Pilchie** [comment]
Also tagging @kuhlenh and @jasonmalinowski so they see this issue.

**Comment by jasonmalinowski** [comment]
We had discussed allowing globbing in .editorconfig for things like:

    dotnet_analyzer.S* = error

but indeed that's a bit icky here since the rule IDs aren't quite, well, unique enough, for my taste in this case.

**Comment by ohadschn** [comment]
@jasonmalinowski that would actually probably cover most cases... 
I assume overrides would be possible right? Something like:

```
dotnet_analyzer.S* = Error
dotnet_analyzer.S123 = Warning
dotnet_analyzer.S456 = None
```

**Comment by jasonmalinowski** [comment]
Yes, we'd allow something like that. My bigger concern is that also means you're potentially turning off any _other_ analyzer that starts with S, such as StyleCop analyzers or things...we might need some syntax for restricting by something else too.

**Comment by ohadschn** [comment]
@jasonmalinowski oh, I thought ```dotnet_analyzer``` in your example was the name of the actual analyzer , so it would only affect that analyzer. If that's not the case, I agree that having the ability to further restrict the rule by ```AnalyzerId``` and/or ```RuleNamespace``` would be preferable.

**Comment by sharwell** [comment]
@mavasani What's the current status of this issue?

**Comment by mavasani** [comment]
I believe this feature request is pretty identical to https://github.com/dotnet/roslyn/issues/16535#issuecomment-427885375, which has already been rejected by the compiler team. The recommendation from compiler team is the users should configure each diagnostic ID, any globbing based support is not going to be added.

**Comment by Kazpers** [comment]
@mavesani

> any globbing based support is not going to be added.

That is extremely unfortunate. Makes the entire system effectively useless. Micromanaging hundreds of individual IDs is a no-go. What was the justification for rejecting it?
